datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -370,135 +369,57 @@ pub fn make_array(arrays: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
-fn return_empty(return_null: bool, data_type: DataType) -> Arc<dyn Array> {
-    if return_null {
-        new_null_array(&data_type, 1)
-    } else {
-        new_empty_array(&data_type)
-    }
-}
-
-fn list_slice<T: Array + 'static>(
-    array: &dyn Array,
-    i: i64,
-    j: i64,
-    return_element: bool,
-) -> ArrayRef {
-    let array = array.as_any().downcast_ref::<T>().unwrap();
-
-    let array_type = array.data_type().clone();
+pub fn array_element(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
Does this need to be `pub`? If it does, could you please add a documentation comment explaining what the arguments are (namely that the first is a list and the second is an array of indexes?

Reply:
Yes, there is array_element sql function